### PR TITLE
[ML] Add e2e test for local command job

### PR DESCRIPTION
**Pull request by TonyJ1**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/dev_requirements.txt [hunk 1]
```diff
@@ -13,5 +13,6 @@ pytest
 pydash
 azure-mgmt-msi
 pywin32==304 ; sys_platform == 'win32'
+docker;platform.python_implementation!="PyPy"
```
Comment (TonyJ1) [inline sdk/ml/azure-ai-ml/dev_requirements.txt:16]: As local_job test works only for win, install only for win?

File: sdk/ml/azure-ai-ml/tests/command_job/e2etests/test_command_job.py [hunk 2]
```diff
@@ -204,9 +205,12 @@ def test_command_job_builder(self, data_with_2_versions: str, client: MLClient)
         len(new_result.inputs) == 3
         assert result.display_name == new_result.display_name
 
-    @pytest.mark.skip(reason="Task 1791832: Inefficient, causing testing pipeline to time out.")
     @pytest.mark.timeout(900)
     @pytest.mark.e2etest
+    @pytest.mark.skipif(
+        condition=not sys.platform.startswith(("win32", "cygwin")),
```
Comment (TonyJ1) [inline sdk/ml/azure-ai-ml/tests/command_job/e2etests/test_command_job.py:211]: 1. As local_job test works only for win, keep only win32 in the list? 
2. Add another condition to exclude pypy?

File: sdk/ml/azure-ai-ml/tests/command_job/e2etests/test_command_job.py [hunk 3]
```diff
@@ -204,9 +205,12 @@ def test_command_job_builder(self, data_with_2_versions: str, client: MLClient)
...
+        reason="Skipping for PyPy as docker installation is not supported and skipped in dev_requirement.txt",
```
Comment (TonyJ1) [inline sdk/ml/azure-ai-ml/tests/command_job/e2etests/test_command_job.py:212]: Create a task to enable local test for non-win systems and update it in the reason.

**Review by luigiw — APPROVED** [review]

File: sdk/ml/azure-ai-ml/tests/command_job/e2etests/test_command_job.py (hunk 3 above)
Comment (TonyJ1) [reply sdk/ml/azure-ai-ml/tests/command_job/e2etests/test_command_job.py:212]: [Task 2287380](https://msdata.visualstudio.com/Vienna/_workitems/edit/2287380): [SDK - LocalJob] E2E test fails for Linux, Mac & PYPI

**Pull request merged**